CMakeLists.txt

```diff
@@ -18,17 +17,6 @@ project("Woof"
 
 set(CMAKE_C_STANDARD 99)
 
-# Set a default build type if none was specified
```

Comment:
I don't see how this can't work?

Reply:
It seems that this condition `if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)` is never true, at lest on Windows.

I don't understand why this code is needed, because of it I thought I had a default `RelWithDebInfo` configuration until I started experimenting with high resolutions and profiling. I think this code is misleading.

Reply:
My point is that I *always* forget to call cmake with `-DCMAKE_BUILD_TYPE=...` and I hate cmake for requiring this. So, I want it to select the most reasonable default by itself if I don't explicitly choose otherwise (which is what *every* program should *always* do IMHO).

Reply:
It's `Debug` by default. I tried WSL Ubutnu and Debian in VM, it's debug there too. I have never seen such code in other projects.

Reply:
I'd prefer it to be `RelWithDebInfo` by default. But, well, so be it.

Reply:
Wait, I checked again - it works on Ubuntu. On Windows, `CMAKE_CONFIGURATION_TYPES` is always defined, even if the non-visual Studio generator is selected. I will restore this code then.

Reply:
Eh, I guess we need to rewrite it: https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-configurations

Reply:
Restored here: https://github.com/fabiangreffrath/woof/pull/1350/commits/c0ae4dfb48a2cab33e21eb03950979bb327a115c

Reply:
BTW, this is the origin of these lines:
https://www.kitware.com/cmake-and-the-default-build-type/